### PR TITLE
Adding appCatalog CRD validation

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/5407

Adding AppCatalog validation.

### CODE REVIEW

File: pkg/apis/application/v1alpha1/app_catalog_types.go [hunk 1]
```diff
@@ -1,10 +1,87 @@
...
+            storage:
+              type: object 
+              properties:
+                type:
+                  type: string
+                URL:
+                  type: string
+                  format: uri 
+              required: ["type", "URL"]
+          required: ["title", "description", "storage"]
```
Comment (tomahawk28) [inline pkg/apis/application/v1alpha1/app_catalog_types.go]: I think logoURL is not that important, right?

Comment (rossf7) [reply pkg/apis/application/v1alpha1/app_catalog_types.go]: Yes it's for display purposes and maybe not needed on internal catalogs.

File: pkg/apis/application/v1alpha1/app_catalog_types.go [hunk 2]
```diff
@@ -1,10 +1,87 @@
...
+spec:
+  group: application.giantswarm.io
+  scope: Cluster
+  version: v1alpha1
+  names:
+    kind: AppCatalog
+    plural: appcatalogs
+    singular: appcatalog
+  subresources:
+    status: {}
```
Comment (rossf7) [inline pkg/apis/application/v1alpha1/app_catalog_types.go]: This isn't enabled in the current CRD. But I don't think it does any harm to add it.

**Review by rossf7 — APPROVED** [review]
LGTM when tested. 

Please make sure this plays nicely with `opsctl ensure appcatalogs`.

**Comment by tomahawk28** [comment]
Tested on ginger. ```opsctl ensure appcatalogs --installation ginger -b appcatalog_validation --user jhbok
Ensuring app catalogs for `ginger`
E0902 14:27:40.523455   37405 portforward.go:376] error copying from local connection to remote stream: tls: use of closed connection
Ensured app catalogs for `ginger`.
Finished.
...

```

**Pull request merged**